sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClient.java

```diff
@@ -1249,25 +1242,45 @@ private boolean isValidKeyLocallyAvailable() {
         }
 
         if (key == null && implClient.getKeyCollection() != null) {
-            key = CryptographyUtils.SECRETS_COLLECTION.equals(implClient.getKeyCollection())
-                ? getSecretKey() : getKey().getKey();
+            try {
+                // Get key from service and validate it. Then attempt to create a local cryptography client or default
+                // to service.
+                key = CryptographyUtils.SECRETS_COLLECTION.equals(implClient.getKeyCollection())
+                    ? implClient.getSecretKey() : getKey().getKey();
+            } catch (HttpResponseException e) {
+                if (e.getResponse().getStatusCode() == 403) {
+                    localOperationNotSupported = true;
+                    
+                    LOGGER.warning("Could not retrieve key from service to perform cryptographic operations locally. "
+                        + "Will use service-side cryptography insetad.", e);
+
+                    return false;
+                } else {
+                    LOGGER.error("Could not retrieve key from service.");
+
+                    throw LOGGER.logExceptionAsError(e);
```

Comment:
This logs the error twice.

Reply:
It logs an ERROR message and then logs the exception.